tools/lsp/common/token_info.rs

```diff
@@ -186,6 +191,62 @@ pub fn token_info(document_cache: &common::DocumentCache, token: SyntaxToken) ->
                 }
                 return find_property_declaration_in_base(document_cache, element, &prop_name);
             }
+            if parent.kind() == SyntaxKind::Component {
+                let doc = document_cache.get_document_for_source_file(&node.source_file)?;
+                let element_type = doc
+                    .local_registry
+                    .lookup_element(
+                        i_slint_compiler::parser::normalize_identifier(token.text()).as_str(),
+                    )
+                    .ok()?;
+                match &element_type {
+                    ElementType::Component(component)
+                        if component
+                            .node
+                            .as_ref()
+                            .map(|n| n.text_range().contains_range(token.text_range()))
+                            .unwrap_or_default() =>
+                    {
+                        return Some(TokenInfo::ElementType(element_type));
+                    }
+                    _ => { /* nothing to do */ }
```

Comment:
Should that be a `if let`

Reply:
Yeap, that is probably nice, the next push will have that.